neqo-http3/src/stream_type_reader.rs

```diff
@@ -20,42 +20,33 @@ impl NewStreamTypeReader {
             fin: false,
         }
     }
+
     pub fn get_type(&mut self, conn: &mut Connection, stream_id: u64) -> Option<u64> {
         // On any error we will only close this stream!
-        loop {
```

Comment:
I think we need to leave a loop here. I am 99% sure. If a stream type is encoded on more than 1 byte we will never read more than the first byte. Do we have a tests for this case?

Reply:
I don't know about tests but I've spent some more time understanding how the decoder works, and agree we need the loop. 

(For anyone else reading this, it's because `IncrementalDecoder::min_remaining()` changes how much it is requesting, depending on what stage of decoding it is in.)